### ISSUE TITLE
Effect checker crashes

### ISSUE THREAD

**Issue by p-offtermatt** [informalsystems/quint#1183]
Loading this spec
https://github.com/cosmos/interchain-security/blob/b992f8c305057d38482c0dd54006336fbb3d0364/tests/difference/core/quint_model/ccv.qnt
with VSCode crashes the effect checker.

Running `quint typecheck ccv.qnt` I get this error:
```
Error: Couldn't find any result for id 1646 while Inferring effect for ConsumerChains
    at EffectInferrer.fetchResult (/opt/homebrew/lib/node_modules/@informalsystems/quint/dist/src/effects/inferrer.js:262:19)
    at EffectInferrer.effectForName (/opt/homebrew/lib/node_modules/@informalsystems/quint/dist/src/effects/inferrer.js:278:25)
    at EffectInferrer.exitName (/opt/homebrew/lib/node_modules/@informalsystems/quint/dist/src/effects/inferrer.js:99:41)
    at walkExpression (/opt/homebrew/lib/node_modules/@informalsystems/quint/dist/src/ir/IRVisitor.js:319:25)
    at /opt/homebrew/lib/node_modules/@informalsystems/quint/dist/src/ir/IRVisitor.js:336:38
    at Array.forEach (<anonymous>)
    at walkExpression (/opt/homebrew/lib/node_modules/@informalsystems/quint/dist/src/ir/IRVisitor.js:336:23)
    at /opt/homebrew/lib/node_modules/@informalsystems/quint/dist/src/ir/IRVisitor.js:336:38
    at Array.forEach (<anonymous>)
    at walkExpression (/opt/homebrew/lib/node_modules/@informalsystems/quint/dist/src/ir/IRVisitor.js:336:23)
```

**Comment by p-offtermatt** [comment]
https://github.com/cosmos/interchain-security/blob/55ab5955d71c4e5ea783cdd1a9a2a9de84cad797/tests/difference/core/quint_model/ccv.qnt#L994
Another spec where the effect crasher crashes, as does 
```$ quint typecheck ccv.qnt
quint typecheck <input>

check types and effects of a Quint specification

Options:
  --help     Show help                                                 [boolean]
  --version  Show version number                                       [boolean]
  --out      output file (suppresses all console output)                [string]

Error: Couldn't find any result for id 1466 while Inferring effect for UnbondingPeriodPerChain
    at EffectInferrer.fetchResult (/Users/offtermatt/projects/quint/quint/dist/src/effects/inferrer.js:262:19)
    at EffectInferrer.effectForName (/Users/offtermatt/projects/quint/quint/dist/src/effects/inferrer.js:278:25)
    at EffectInferrer.exitName (/Users/offtermatt/projects/quint/quint/dist/src/effects/inferrer.js:99:41)
    at walkExpression (/Users/offtermatt/projects/quint/quint/dist/src/ir/IRVisitor.js:319:25)
    at /Users/offtermatt/projects/quint/quint/dist/src/ir/IRVisitor.js:336:38
    at Array.forEach (<anonymous>)
    at walkExpression (/Users/offtermatt/projects/quint/quint/dist/src/ir/IRVisitor.js:336:23)
    at /Users/offtermatt/projects/quint/quint/dist/src/ir/IRVisitor.js:336:38
    at Array.forEach (<anonymous>)
    at walkExpression (/Users/offtermatt/projects/quint/quint/dist/src/ir/IRVisitor.js:336:23)
```

**Comment by bugarela** [comment]
Hey! Thanks again for the amazing report with multiple examples. I was able to find the issue by debugging your commit hashes :smile: 

> so it seems to be an issue of using the instantiated constants of the instanced module in the main module?

Pretty much! Turns out our topological sorting of declarations was not considering the overrides inside instance statements (i.e. `ConsumerChains = consumerChains`), so the effect checker was trying to compute effects for the name `ConsumerChains` before inferring effects from the instance statement. I have a fix for the toposort that makes the declarations be visited in the correct order, and then the effect checker doesn't crash anymore.
